### ISSUE TITLE
Inquiring About Loss

### ISSUE THREAD

**Issue by kailugaji** [fuyw/FuRL#6]
Hello, thank you for your excellent paper. I apologize for bothering you again. Due to my weak foundation in mathematics, I don't understand the correspondence between Eq. (4) in the original paper and the code. In line 101 of FuRL/models/projection.py, why is the expression of loss "neg_loss = neg_mask * (neg_cosine - pos_cosine)" multiplied by (neg_cosine - pos_cosine)?

![1](https://github.com/user-attachments/assets/06633b52-837f-4407-b945-644a4b2c67bc)

**Comment by fuyw** [comment]
The loss is the same as  https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html.

**Comment by kailugaji** [comment]
> The loss is the same as https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html.

Thank you for your reply.

**Comment by kailugaji** [comment]
> The loss is the same as https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html.

neg_mask can be considered as MarginRankingLoss, but why is it still necessary to multiply it by (neg_cosine - pos_cosine) on this basis? thank you.

**Comment by fuyw** [comment]
neg_mask is the relu operator and (neg_cosine - pos_cosine) is the loss value

**Comment by kailugaji** [comment]
> neg_mask is the relu operator and (neg_cosine - pos_cosine) is the loss value

Thank you very much~
